wdl-format/src/comments.rs

```diff
@@ -0,0 +1,192 @@
+                                comments_found = true;
+
+                                if would_be_interrupting {
+                                    formatter.interrupt();
+                                }
+
+                                let comment = Comment::cast(
+                                    cur.as_token().expect("Comment should be a token").clone(),
+                                )
+                                .expect("Comment should cast to a comment");
+
+                                comment.format(&mut inner_buffer, formatter)?;
+                                if inner_buffer.starts_with("## ") {
+                                    inner_buffer.remove(0);
+                                }
+                                reversed_text.push(inner_buffer.clone());
+                                inner_buffer.clear();
+                            }
+                        }
+                        _ => {
+                            // The 'cur' comment is on the same line as this
+                            // token. It "belongs"
+                            // to whatever is currently being processed.
+                        }
+                    }
+                }
+            }
+            SyntaxKind::Whitespace => {
+                // Ignore whitespace until a comment is found.
+                // Ignore whitespace that doesn't contain at least two newlines.
+                // (Two newlines indicates a blank line). Since each comment is
+                // followed by a newline, we only insert one newline here.
+                if cur.to_string().matches(NEWLINE).count() > 1 && comments_found {
+                    inner_buffer.push_str(NEWLINE);
+                    reversed_text.push(inner_buffer.clone());
+                    inner_buffer.clear();
+                }
+            }
+            _ => {
+                // We've backed up to non-trivia, so we can stop
+                break;
+            }
+        }
+        prev = cur.prev_sibling_or_token()
+    }
+
+    if comments_found && would_be_interrupting && !began_interrupted {
+        write!(writer, "{}", NEWLINE)?;
+    }
+
+    // Skip any whitespace before comments start.
+    let mut comment_processed = false;
+    for line in reversed_text.iter().rev() {
+        if line.contains('#') {
+            comment_processed = true;
+            formatter.indent(writer)?;
+        }
+        if comment_processed {
+            write!(writer, "{}", line)?;
+        }
+    }
+    Ok(())
+}
+
+/// Format a comment on the same line as an element.
+///
+/// If 'would_be_interrupting' is false and there is no comment on the line, a
+/// newline will be inserted.
+pub fn format_inline_comment<T: std::fmt::Write>(
+    element: &SyntaxElement,
+    writer: &mut T,
+    formatter: &mut Formatter,
+    would_be_interrupting: bool,
+) -> std::fmt::Result {
+    let mut next = element.next_sibling_or_token();
+    while let Some(cur) = next {
```

Comment:
You can just do this, avoid the line above and the setting at the end of the
`while` loop.

```suggestion
    while let Some(cur) = element.next_sibling_or_token() {
```

Reply:
Having `element` in the loop condition doesn't work, as `cur` will be perpetually assigned the same value. I believe the "extra" variable (`next`) is needed here for the `while let` logic. Maybe there's a different loop that can be used?